### PR TITLE
feat(QPT):QE-187 Expand "I" object to include common helper methods.

### PR DESCRIPTION
**Pull request by BC-Ubalderas**
Jira: [QE-187](https://jira.bigcommerce.com/browse/QE-187)

## What? Why?
Changed the steps file, so that the customSteps got the methods from the page and helper objects, so that the framework "I" object inherited their methods, making the specs more user behavior centric.

Spent a lot of time looking into converting page objects and helpers into classes, but ended up going through using the Object.assign method instead, to copy methods from them into the customSteps objects that gets passed into an actor() type object on the steps file, which seemed the most elegant form of this conversion. There are a ton of other ways to do this, but this seemed the less intrusive into keeping page objects and helpers/step objects stay as close as possible to the recommendations and best practices per the documentation of CodeceptJS.

## Affected areas?
All the tests were affected, but only in a minor way.

## Additional Deploy Steps? (Optional)
None

## How was it tested?
Ran all the functional and unit tests, had to modify them for them to be able to use the I object instead. 

## Testing / Proof

Screen Shot 2020-09-18 at 7 18 43 PM

## How can this be undone?
1. Revert this PR

ping @bigcommerce/billing-iam-testing

### CODE REVIEW

**Comment by avattipalli** [comment]
Looks very clean! LGTM for the most part but I will take a second look at it between the meetings today and 👍 
Thanks for working on this @BC-Ubalderas

File: .env-example [hunk 1]
```diff
@@ -10,3 +10,6 @@ STAGING_LOGIN_URL='https://login.staging.zone/'
 # Test User credentials - for local runs and troubleshooting
 TEST_USER_EMAIL='some_test_user_email@bigcommerce.com'
 TEST_USER_PASSWORD='custom_password'
+
+# Headless variable for easy configuration on local test runs
```
Comment (avattipalli) [inline .env-example:14]: I don't think a comment is necessary here, but totally up to you.

File: test/functional/billing/account-spec.js [hunk 2]
```diff
@@ -2,20 +2,20 @@ import {BMP_HOST, TEST_USER_EMAIL, TEST_USER_PASSWORD} from '../../../src/global
...
 
 Scenario('Merchant with trial store can log into account management interface', (I) => {
     I.see('My Account');
 });
 
 Scenario('Merchant with trial store can change company name', (I, bmpClientHeaderNav, bmpAccountHelper) => {
-    bmpClientHeaderNav.navigateToAccountOptions('Account details');
-    bmpAccountHelper.editAccountInformation('BMP Test Account')
+    I.navigateToAccountOptions('Account details');
+    I.editAccountInformation('BMP Test Account')
```
Comment (avattipalli) [inline test/functional/billing/account-spec.js:20]: missing `;`

File: test/functional/billing/account-spec.js [hunk 3]
```diff
@@ -2,20 +2,20 @@ import {BMP_HOST, TEST_USER_EMAIL, TEST_USER_PASSWORD} from '../../../src/global
...
-    bmpLogoutHelper.logoutFromBMPCUI();
+After((I) => {
+    I.logoutFromBMPCUI();
 });
 
 Scenario('Merchant with trial store can log into account management interface', (I) => {
     I.see('My Account');
 });
 
 Scenario('Merchant with trial store can change company name', (I, bmpClientHeaderNav, bmpAccountHelper) => {
```
Comment (avattipalli) [inline test/functional/billing/account-spec.js:18]: We can remove `bmpClientHeaderNav ` and `bmpAccountHelper ` 🍺

File: test/functional/billing/account-spec.js (hunk 2 above)
Comment (BC-Ubalderas) [reply test/functional/billing/account-spec.js]: Thanks, fixed it.

File: .env-example (hunk 1 above)
Comment (aaron-humerickhouse) [reply .env-example:14]: Maybe explain what changing it to false would do?  My guess is people don't know the tooling and methods

File: test/functional/billing/account-spec.js (hunk 3 above)
Comment (BC-Ubalderas) [reply test/functional/billing/account-spec.js]: Done.

**Review by avattipalli — APPROVED** [review]
LGTM!

One question I have for this change is if we want to add documentation along with this PR to guide someone new to repo what's the expected path to add a helper so that it gets included in `I` OR we can expand a little more on that and have a separate ticket to add the documentation. I'm fine with either.

Let's wait for another 👍  and get this merged!

File: .env-example (hunk 1 above)
Comment (BC-Ubalderas) [reply .env-example]: Added a more explicit comment.

File: test/functional/common/cp-side-nav-spec.js [hunk 4]
```diff
@@ -2,15 +2,15 @@ import {CP_HOST, TEST_USER_EMAIL, TEST_USER_PASSWORD} from '../../../src/global-
...
 
-After(( loginHelper) => {
-    loginHelper.logoutFromUI();
+After(( I ) => {
+    I.logoutFromUI();
 });
 
-Scenario('validate cp-side-nav-page', ( cpSideNavPage) => {
-    cpSideNavPage.validateCpSideNavPage();
+Scenario('validate cp-side-nav-page', ( I) => {
```
Comment (j05h) [inline test/functional/common/cp-side-nav-spec.js:14]: Spacing around the `I`

File: test/functional/common/user-settings-spec.js [hunk 5]
```diff
@@ -2,22 +2,22 @@ import {CP_HOST, TEST_USER_EMAIL, TEST_USER_PASSWORD} from '../../../src/global-
...
+Scenario('validate user-settings page', (I) => {
+    I.validateUserSettingsPage();
 });
 
-Scenario('validate user-settings-helper', ( userSettingsHelper) => {
-    userSettingsHelper.navigateToUserSettings();
-    userSettingsHelper.navigateToEditProfile();
-    userSettingsHelper.navigateToChangeEmailAddress();
-    userSettingsHelper.navigateToChangePassword();
+Scenario('validate user-settings-helper', ( I) => {
```
Comment (j05h) [inline test/functional/common/user-settings-spec.js:18]: Spacing around the `I`

File: test/functional/common/cp-side-nav-spec.js (hunk 4 above)
Comment (BC-Ubalderas) [reply test/functional/common/cp-side-nav-spec.js]: Fixed.

File: test/functional/common/user-settings-spec.js (hunk 5 above)
Comment (BC-Ubalderas) [reply test/functional/common/user-settings-spec.js]: Fixed.

**Review by MarkJCram — APPROVED** [review]
This looks good and addresses all issues we have previously spoken about.  I'm also including in documentation some of the concerns raised surrounding this utility change

**Pull request merged**